src/viam_mlmodelservice_triton_impl.cpp

```diff
@@ -103,6 +103,12 @@ class Service : public vsdk::MLModelService, public vsdk::Stoppable, public vsdk
             std::lock_guard<std::mutex> lock(state_lock_);
             if (!stopped_) {
                 stopped_ = true;
+
+                // Remove any symlinks we've added to our module workspace.
+                if (!state_->model_name.empty()) {
```

Comment:
I think it is impossible to succeed to configure the module with a missing or empty `model_name`, so I think you don't need the check:
```        // Pull the model name out of the configuration.
        auto model_name = attributes->find("model_name");
        if (model_name == attributes->end()) {
            std::ostringstream buffer;
            buffer << service_name
                   << ": Required parameter `model_name` not found in configuration";
            throw std::invalid_argument(buffer.str());
        }

        auto* const model_name_string = model_name->second->get<std::string>();
        if (!model_name_string || model_name_string->empty()) {
            std::ostringstream buffer;
            buffer << service_name
                   << ": Required non-empty string parameter `model_name` is either not a "
                      "string "
                      "or is an empty string";
            throw std::invalid_argument(buffer.str());
        }
        state->model_name = std::move(*model_name_string);
```

Reply:
Suppose `reconfigure` never succeeds. Couldn't `stop` still be called during shutdown? and in that case, if `model_name` is empty, I'd rather not delete the entire `$VIAM_MODULE_DATA` directory.

It's possible the check is redundant, but I don't think that's obvious. I'd prefer to leave it in, as a defensive programming measure.